src/lightning/fabric/utilities/seed.py

```diff
@@ -7,13 +7,18 @@
 
 import numpy as np
 import torch
+from lightning_utilities.core.imports import RequirementCache
 
 from lightning.fabric.utilities.rank_zero import _get_rank, rank_prefixed_message, rank_zero_only, rank_zero_warn
 
 log = logging.getLogger(__name__)
+_NUMPY_AVAILABLE = RequirementCache("numpy")
 
-max_seed_value = np.iinfo(np.uint32).max
-min_seed_value = np.iinfo(np.uint32).min
+if _NUMPY_AVAILABLE:
+    import numpy as np
+
+    max_seed_value = np.iinfo(np.uint32).max
+    min_seed_value = np.iinfo(np.uint32).min
```

Comment:
I suggest to actually just write these values as Python integers. We can write in a comment how we computed them:

```py
max_seed_value = 4294967295  # 2^32 - 1 (uint32)
min_seed_value = 0
```